File: app/lib/models/item.dart
class Item {
  int id;
  String name;
  int quantity;
  int storageID;

  int targetQuantity;
  String details;
  String barCode;

  Item({
    required this.id,
    required this.name,
    required this.quantity,
    required this.storageID,
    required this.targetQuantity,
    required this.details,
    required this.barCode,
  });

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'quantity': quantity,
      'storageID': storageID,
      'targetQuantity': targetQuantity,
      'details': details,
      'barCode': barCode,
    };
  }

  Item.fromJson(Map<String, dynamic> json)
      : id = json['id'] as int,
        name = json['name'] as String,
        quantity = json['quantity'] as int,
        storageID = json['storageID'] as int,
        targetQuantity = json['targetQuantity'] != null ? json['targetQuantity'] as int : 0,
        details = json['details'] != null ? json['details'] as String : "",
        barCode = json['barCode'] != null ? json['barCode'] as String : "";
}

File: backend/internal/types/item.go
package types

type Item struct {
	ID             int    `json:"id"`
	StorageID      int    `json:"storageID"`
	Name           string `json:"name"`
	Quantity       int    `json:"quantity"`
	TargetQuantity int    `json:"targetQuantity"`
	Details        string `json:"details"`
	Barcode        string `json:"barcode"`
}

type PostItemRequest struct {
	UserAccountID  int    `json:"userAccountID"`
	StorageID      int    `json:"storageID"`
	Name           string `json:"name"`
	Quantity       int    `json:"quantity"`
	TargetQuantity int    `json:"targetQuantity"`
	Details        string `json:"details"`
	Barcode        string `json:"barcode"`
}

type PatchItemRequest struct {
	UserAccountID  int    `json:"userAccountID"`
	ID             int    `json:"id"`
	OldStorageID   int    `json:"oldStorageID"`
	StorageID      int    `json:"storageID"`
	Name           string `json:"name"`
	Quantity       int    `json:"quantity"`
	TargetQuantity int    `json:"targetQuantity"`
	Details        string `json:"details"`
	Barcode        string `json:"barcode"`
}

type DeleteItemRequest struct {
	UserAccountID int `json:"userAccountID"`
	ID            int `json:"id"`
}
